### ISSUE TITLE
Last rportd binary is unusable on linux 

### ISSUE THREAD

**Issue by buzzzo** [IOTech17/neo-rport#2]
I've downloaded rportd_9.12_linux_x86_64.tar.gz
Untarred but when i'm trying to run it I receive:

root@rport1 rport]# /usr/src/rport/rportd
-bash: /usr/src/rport/rportd: No such file or directory

The old binary works:

/usr/local/bin/rportd
2023/10/04 10:57:41 Invalid config: client authentication must be enabled: set either 'auth', 'auth_file' or 'auth_table'. See --help

**Comment by IOTech17** [comment]
@buzzzo 

I started by investigating the file it self : 

```
file rportd

rportd: ELF 64-bit LSB executable, x86-64, version 1 (SYSV), dynamically linked, interpreter /lib/ld-musl-x86_64.so.1, Go BuildID=HU95XhuLwXoEtzxoepRx/-b-bV1jSpS_Ggwy7hpUl/r45JNkELCyfQlvvw1JhR/ZH5KPudJQXVYcjBCL9u-, stripped
```

As you can see it needs musl in order to work.

I solved the issue by installing musl (apt install musl)

Then when running the file I got : 

```
./rportd

Invalid config: client authentication must be enabled: set either 'auth', 'auth_file' or 'auth_table'. See --help
```

**Comment by buzzzo** [comment]
Thx, i'm on almalinux/rocky so in order to install lib-musl should use an external repo:

ownload cert-forensics-tools-release-el8 rpm:
https://forensics.cert.org/cert-forensics-tools-release-el8.rpm
Install cert-forensics-tools-release-el8 rpm:
# rpm -Uvh cert-forensics-tools-release*rpm
Install musl-libc rpm package:
# dnf --enablerepo=forensics install musl-libc

**Comment by buzzzo** [comment]
I've version: 0.9.5
when try to run my instance with the new bin:

2023/10/05 11:39:13 failed to init Client Repository: failed to get clients: sql: Scan error on column index 3, name "details": failed to decode 'details' field: json: cannot unmarshal object into Go struct field CommandsConfig.client_configuration.remote_commands.allow_regexp of type *regexp.Regexp

Could be something related to some changes in sql schema.
Obviously I can't upgrade cause some genius has wiped all the old docs.

Thx

**Comment by IOTech17** [comment]
I haven't change anything on the SQL format, I simply updated for now the link to the go module import and corrected the goreleaser flag when compiling the binary with docker-goreleaser but that's it

Are you loading a conf file when running rportd?

**Comment by buzzzo** [comment]
Of course i run the full command line.

I'm suspecting there are some sql changes between versions.
Of course without any additional informations (eg: upgrade notes) it's quite impossible to upgrade the sql.

Thx

**Comment by IOTech17** [comment]
Regarding the SQL I haven't changed anything it is like the latest rport version (9.12)  which DB are you using sqlite or mysql?

**Comment by buzzzo** [comment]
I'm using sqlite

**Comment by IOTech17** [comment]
I can see that you are trying to upgrade from Rport 0.9.5 to 9.12.

I know that they made some changes in the db schema in the latest version.

**Comment by buzzzo** [comment]
The issue is related with the table client ,which is:

CREATE TABLE clients (
    id TEXT PRIMARY KEY NOT NULL,
	client_auth_id TEXT NOT NULL,
    disconnected_at DATETIME,
	details TEXT NOT NULL
) WITHOUT ROWID;


so the filed details is present, and go complains about ommandsConfig.client_configuration.remote_commands.allow_regexp. but i can't find anything related on rportd.conf

**Comment by buzzzo** [comment]
i've tried to decode the error:
2023/10/09 12:23:34 failed to init Client Repository: failed to get clients: sql: Scan error on column index 3, name "details": failed to decode 'details' field: json: cannot unmarshal object into Go struct field CommandsConfig.client_configuration.remote_commands.allow_regexp of type *regexp.Regexp
i think it been trying to decode the field "details" and load in to a json object
but fails because something is wrong in the json load in some records of clients table
something that ha broken json in details field 
something related to the remote command.allow_regexp
for example this is an excerpt from a details field:
"remote_commands":{"enabled":true,"send_back_limit":4194304,"allow":[".*"],"deny":[],"order":["allow","deny"],"allow_regexp":[{}],"deny_regexp":[]},"
so i think something is changed from 0.9.5 to 0.9.12 in the management of db

**Comment by buzzzo** [comment]
FOLLOWUP: in the i figure out the issue
i've empty the db clients
and load ONLY one clients
with the remote_connect modified
from allow_regexp":[{}],"deny_regexp":[]},"
to allow_regexp":[],"deny_regexp":[]},"
and the new rportd bin started

**Comment by IOTech17** [comment]
@buzzzo 

Great that you figured it out.

I know that at the time I was using a mysql DB and was experiencing similar issue when upgrading to the version 0.9.12

**Comment by buzzzo** [comment]
I think the sql backend does not matter.
details field is simply a storage for client config, just a text blob where uploading client's info.

Just to know: How do you have solved the issue ?

**Comment by IOTech17** [comment]
the technical documentation of rport was lacking in regard to DB schema, and the command needed for the DB table to be created.

On my side when it happened I remove the database and started from scratch, simply used the same password for the different client group.

**Comment by buzzzo** [comment]
And so the clients start re-register itself without the need to re-add manually ?

**Comment by IOTech17** [comment]
yes should come back automatically as long as you keep the same ID and password.

I am going to remove this repository and continue working on the openrport repository, they manage to have all the necessary services running (KB, pairing system etc...)
